signalfx-java/src/main/java/com/signalfx/signalflow/Channel.java

```diff
@@ -0,0 +1,97 @@
+package com.signalfx.signalflow;
+
+import java.io.Closeable;
+import java.util.Iterator;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Copyright (C) 2016 SignalFx, Inc. All rights reserved.
+ * 
+ * Abstract immutable representation for open channels that receive streaming data from a SignalFlow
+ * computation.
+ * 
+ * Channel objects bridge the gap between an underlying transport and a higher-level Computation
+ * object by providing a transport-agnostic and encoding-agnostic access to the stream of
+ * messages.StreamMessage objects that are received for a given computation.
+ * 
+ * Channels are iterable that return ChannelMessage instances.
```

Comment:
I don't typically add author to anything that isn't frozen for eternity .. but can do it.